src/data/slices.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-param-reassign,import/prefer-default-export */
+import { createSlice } from '@reduxjs/toolkit';
+
+import { RequestStatus } from './constants';
+
+const blocksSlice = createSlice({
+  name: 'courseBlocks',
+  initialState: {
+    status: RequestStatus.IN_PROGRESS,
+    topics: {
+      // Maps topic id to the discussion course path and link
+    },
+    // List of chapter blocks in the course. Rest of the structure can be derived from children of topics
+    chapters: [],
+    // Mapping of block keys to block data
+    blocks: {},
+  },
+  reducers: {
+    fetchCourseBlocksRequest: (state) => {
+      state.status = RequestStatus.IN_PROGRESS;
+    },
+    fetchCourseBlocksSuccess: (state, { payload }) => {
+      state.status = RequestStatus.SUCCESSFUL;
+      Object.assign(state, payload);
```

Comment:
is it equivalent to? `state = [...state, ...payload]`

Reply:
yes, both are the same only difference is syntax. the one you mentioned is standardized for reducer but the application of `object.assign` can also be seen in case of complex nested data structure.

Reply:
They are not exactly equivalent, and in this case you can't use the syntax above. 

``state = { ...state, ...payload }`` is equivalent to ``state = Object.assign({}, state, payload}``

i.e. the spread syntax creates a new object and copies over properties from both state and payload onto the new object. `Object.assign(state, payload)` on the other hand copies over properties from `payload` to `state`. 

The reason this is important here is that if you assign another value to `state` then it won't be tracked. If you update state by changing a property (i.e. `state.xyz = abc`) then that will update the original state. However, if you set state to a new value entirely then the original `state` object will be untouched since state just refers to a new object inside the reducer now and that can't be tracked.

Reply:
thank you @xitij2000